core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -912,22 +947,43 @@ private void updateSchema(final GraphConfig config) {
                         .build();
                 addSchema(newSchema);
             }
+
+            if (null != config.getLibrary() && config.getLibrary().exists(config.getGraphId())) {
+                // Set Schema if null.
+                final Pair<Schema, StoreProperties> pair = config.getLibrary().get(config.getGraphId());
+                schema = (null == schema) ? pair.getFirst() : schema;
+            }
         }
 
-        private void updateStoreProperties(final GraphConfig config) {
-            StoreProperties mergedStoreProperties = null;
+        private void updateStoreProperties(final GraphConfig config, final Store store) {
+            final StoreProperties mergedStoreProperties = new StoreProperties();
+
+            //Add Properties from ParentProperties
             if (null != parentStorePropertiesId) {
-                mergedStoreProperties = config.getLibrary().getProperties(parentStorePropertiesId);
+                mergedStoreProperties.merge(config.getLibrary().getProperties(parentStorePropertiesId));
             }
 
+            //Add Properties from Builder
```

Comment:
Redundant null check as the `merge` method already checks for it [see](https://github.com/gchq/Gaffer/blob/develop/core/store/src/main/java/uk/gov/gchq/gaffer/store/StoreProperties.java#L262)